no-used-before-define is erroneously throw when destructuring object with rest operator

I think this one is legit :)
### How to reproduce
Using `babel-eslint@5.0.0-beta10`
**.eslintrc**
```json
{
    "parser": "babel-eslint",
    "env": {
        "es6": true,
    },
    "parserOptions": {
        "ecmaFeatures": {
            "experimentalObjectRestSpread": true
        },
    },
    "rules": {
        "no-use-before-define": 2,
    }
}
```
**test.js**
```js
const { a, ...b } = { a: 1, b: 2, c:3 };
console.log(a, b);
```

When executing
```shell
$ eslint test.js
```
The linter raise the following error
```
test.js
   2:9   error  'a' was used before it was defined  no-use-before-define
   2:15  error  'b' was used before it was defined  no-use-before-define
```
(nothing appear when using the eslint parser)

--- a/index.js
+++ b/index.js
@@ -320,18 +320,6 @@ function monkeypatch() {
         if (typeAnnotation) {
           checkIdentifierOrVisit.call(this, typeAnnotation);
         }
-        if (id.type === "ObjectPattern") {
-          // check if object destructuring has a spread
-          var hasSpread = id.properties.filter(function(p) {
-            return p._babelType === "SpreadProperty" || p._babelType === "RestProperty";
-          });
-          // visit properties if so
-          if (hasSpread.length > 0) {
-            for (var j = 0; j < id.properties.length; j++) {
-              this.visit(id.properties[j]);
-            }
-          }
-        }
       }
     }
     variableDeclaration.call(this, node);

--- a/test/non-regression.js
+++ b/test/non-regression.js
@@ -1137,18 +1137,45 @@ describe("verify", function () {
     );
   });
 
-  it("visits excluded properties left of spread #95", function () {
+  // This two tests are disabled, as the feature to visit properties when
+  // there is a spread/rest operator has been removed as it caused problems
+  // with other rules #249
+  it.skip("visits excluded properties left of spread #95", function () {
     verifyAndAssertMessages(
       "var originalObject = {}; var {field1, field2, ...clone} = originalObject;",
-      { "no-undef": 1, "no-unused-vars": 1, "no-redeclare": 1 },
+      { "no-unused-vars": 1 },
       []
     );
   });
 
-  it("visits excluded properties left of spread #210", function () {
+  it.skip("visits excluded properties left of spread #210", function () {
     verifyAndAssertMessages(
       "const props = { yo: 'yo' }; const { ...otherProps } = props;",
-      { "no-undef": 1, "no-unused-vars": 1, "no-redeclare": 1 },
+      { "no-unused-vars": 1 },
+      []
+    );
+  });
+
+  it("does not mark spread variables false-positive", function () {
+    verifyAndAssertMessages(
+      "var originalObject = {}; var {field1, field2, ...clone} = originalObject;",
+      { "no-undef": 1, "no-redeclare": 1 },
+      []
+    );
+  });
+
+  it("does not mark spread variables false-positive", function () {
+    verifyAndAssertMessages(
+      "const props = { yo: 'yo' }; const { ...otherProps } = props;",
+      { "no-undef": 1, "no-redeclare": 1 },
+      []
+    );
+  });
+
+  it("does not mark spread variables as use-before-define #249", function () {
+    verifyAndAssertMessages(
+      "var originalObject = {}; var {field1, field2, ...clone} = originalObject;",
+      { "no-use-before-define": 1 },
       []
     );
   });

